PhpcrMigration/Infrastructure/Repository/EntityRepository.php

```diff
@@ -35,19 +35,25 @@ public function insertOrUpdate(array $data, string $tableName, array $types, arr
     {
         $exists = [] !== $where && $this->exists($tableName, $where);
 
-        match ($exists) {
-            true => $this->connection->update(
+        if ($exists) {
+            $this->connection->update(
                 $tableName,
                 $data,
                 $where,
                 $types
-            ),
-            default => $this->connection->insert(
+            );
+        } else {
+            // If this is PostgreSQL and we're doing an insert, we need to handle the ID
+            if ($this->isPostgreSql() && !(isset($data['id']) && isset($data['uuid']))) {
+                $data['id'] = $this->getNextIdValue($tableName);
+            }
```

Comment:
`PostgreSQL` alone is not a indicator that a sequences are used.

Since later doctrine versions sequences are replaced with auto increment id via an option. We probably require a deeper check the underlaying metadata for `id`. In this case as it looks like we don't have the entity manager so maybe we can go over the schema manager. Can you check if `listSequences` works:

https://www.doctrine-project.org/projects/doctrine-dbal/en/4.2/reference/schema-manager.html#listsequences